tools/docker.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2024 Huawei Technologies Co., Ltd. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package tools
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"os/user"
+)
+
+const (
+	docker    = "docker"
+	image     = "ghcr.io/open-s4c/vsyncer"
+	tag       = "latest"
+	useDocker = "false"
+)
+
+func init() {
+	RegEnv("VSYNCER_DOCKER", useDocker, "Use vsyncer Docker container for GenMC, Dat3m, etc")
+}
+
+func DockerRun(ctx context.Context, args []string, volumes []string) error {
+	var (
+		cmd = []string{"run", "--rm"}
+	)
+	// find current user
+	u, err := user.Current()
+
+	// find out current directory
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	// check docker installation
+	// is docker installed?
+	//
+	// are we running outside docker?
+	//
+	// is it rootless?
+	// docker info -f "{{println .SecurityOptions}}" | grep rootless
+	//
+	// if not rooless do I have permission?
+	// check if user in docker group, otherwise should we request sudo?
```

Comment:
Did you plan to add this as part of this MR, or are you leaving it as a todo?

Reply:
Todo